logger/options.go

```diff
@@ -40,43 +39,66 @@ type Options struct {
 	Timestamp     time.Time
 }
 
-func (e InvalidOptionMessage) String() string {
-	return fmt.Sprintf("Options.%s: %s", e.Option, e.Message)
+type fieldIssue struct {
+	field string
+	prob  string
 }
 
-func validateOptionLength(option string, value string, problems *[]string) {
-	if len(value) > 32 {
-		*problems = append(*problems, InvalidOptionMessage{option, "length must be less than 32"}.String())
+type optionsError struct {
+	issues []fieldIssue
+}
+
+func (e *optionsError) Error() string {
+	errorString := ""
+	for i := 0; i < len(e.issues); i++ {
+		errorString += fmt.Sprintf("%s: %s \n", e.issues[i].field, e.issues[i].prob)
+	}
+	return errorString
```

Comment:
Building a string in a loop like this isn't usually very efficient and you'll rely on the compiler / runtime to implicitly optimize it for you which can be hit or miss depending on how complex the logic is. In reality it's probably a non issue as there aren't a lot of possible `fieldIssues` in a single option... but if we're validating the options with every line it might be a worthwhile change.

You might want to consider using Go's String Builder implementation instead.

Reply:
fixed 👍